cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkRuntimeContextProvider.java

```diff
@@ -183,6 +184,7 @@ public void run() {
         injector.getInstance(SecureStore.class),
         injector.getInstance(SecureStoreManager.class)
       );
+      LoggingContextAccessor.setLoggingContext(sparkRuntimeContext.getLoggingContext());
```

Comment:
It is unrelated, but fixes the Spark logging issue (I was debugging with logs and found this issue).